test/puppetlabs/puppetdb/testutils/jetty.clj

```diff
@@ -81,19 +83,22 @@
   ([f] (puppetdb-instance (create-config) f))
   ([config f] (puppetdb-instance config [] f))
   ([config services f]
-   (let [[log-file config] (-> config conf/adjust-tk-config assoc-logging-config)
+   (let [[log-file config] (-> config
+                               conf/adjust-tk-config
+                               assoc-open-port
+                               assoc-logging-config)
          prefix (get-in config
                         [:web-router-service
-                         :puppetlabs.puppetdb.cli.services/puppetdb-service])]
+                         ::svcs/puppetdb-service])]
      (try
        (tkbs/with-app-with-config server
-         (concat [jetty9-service puppetdb-service message-listener-service command-service webrouting-service metrics-service]
+         (concat [jetty9-service svcs/puppetdb-service message-listener-service command-service webrouting-service metrics-service]
                  services)
          config
          (binding [*server* server
                    *base-url* (merge {:protocol "http"
                                       :host "localhost"
-                                      :port (current-port server)}
+                                      :port (get-in config [:jetty :port])}
                                      (when prefix {:prefix prefix}))]
```

Comment:
What happens if the port is taken between (open-port-num) and the server's binding?  As long that's an error, and we don't end up just choosing some other port, then I think we're fine.